packages/content-conformance/src/content-file.ts

```diff
@@ -19,12 +21,33 @@ import type { Node } from 'unist'
 export class ContentFile extends VFile {
   __type = 'content' as const
 
+  constructor(value: Compatible) {
+    super(value)
+    this.parseFrontmatter()
+  }
+
   private tree?: Readonly<Node>
 
   private parseContent() {
     this.tree = remark().use(remarkMdx).parse(this)
   }
 
+  private parseFrontmatter() {
```

Comment:
@thiskevinwang Did we figure out if it's possible to glean errors from the YAML parsing? I'm thinking it would be helpful if we encounter an error during frontmatter parsing, we can attach a message to the file to report on it.

Reply:
The one error I've been able to catch is when using valid but empty fences:

```markdown
---
---

# Hello
```


```
TypeError: Cannot read properties of undefined (reading 'length')
```
- https://github.com/vfile/vfile-matter/blob/main/index.js#L47


Is something like this what you're hinting at?
```typescript
  private parseFrontmatter() {
    try {
      // ...
    } catch (err) {
      this.message(`Error parsing frontmatter: ${err}`)
    }
  }
```

Reply:
Yeah, but I don't think it'll throw. What about this:
```yaml
description: |
this is an invalid multi-line description.
```

what's `this.data.frontmatter` contain?

Reply:
In both my case and your case, `matter()` throws and `this.data.matter` is `undefined`

Reply:
@BRKalow Updated this `parseFrontmatter` with try+catch and added `content-file.test.ts` cases